projects/ngrx.io/content/guide/signals/signal-store/entity-management.md

```diff
@@ -1,3 +1,397 @@
+
+---
+
+Here is the version for updating or removing all entities.
+
+```typescript
+patchState(
+  this.todoStore,
+  addEntities([
+    { id: 2, name: 'Car Washing', finished: false },
+    { id: 3, name: 'Cat Feeding', finished: false },
+  ])
+);
+
+patchState(this.todoStore, updateAllEntities({ finished: true }));
+
+patchState(this.todoStore, removeAllEntities());
+```
+
+For obvious reasons, no id is necessary for operations on all entities.
+
+---
+
+There is also an option to update or remove entities conditionally. We can provide an array of ids or a predicate. The predicate gets the entity as a parameter and needs to return a `boolean`.
+
+```typescript
+patchState(
+  this.todoStore,
+  addEntities([
+    { id: 2, name: 'Car Washing', finished: false },
+    { id: 3, name: 'Cat Feeding', finished: false },
+  ])
+);
+
+patchState(
+  this.todoStore,
+  updateEntities({
+    predicate: (todo) => Boolean(todo.name.match(/cat/i)),
+    changes: { finished: true },
+  })
+);
+
+patchState(
+  this.todoStore,
+  removeEntities((todo) => todo.finished)
+);
+```
+
+For the predicate version, `updateEntities` requires an object literal with `predicate` and `changes`. `removeEntities` works with the simple predicate function.
+
+Note that the predicate function needs to return `false` or `true`. A truthy or falsy value is incorrect.
+
+The alternative version with an array of ids goes like this:
+
+```typescript
+patchState(
+  this.todoStore,
+  addEntities([
+    { id: 2, name: 'Car Washing', finished: false },
+    { id: 3, name: 'Cat Feeding', finished: false },
+    { id: 4, name: 'Dog Feeding', finished: false },
+  ])
+);
+
+patchState(
+  this.todoStore,
+  updateEntities({
+    ids: [2, 3],
+    changes: { finished: true },
+  })
+);
+
+patchState(this.todoStore, removeEntities([2, 4]));
+```
+
+## Customized Id property
+
+The default property name for an identifier is `id` and is of type `string` or `number`.
+
+It is possible to specify a different name, but the type must still be `string` or `number`. You can specify the id only when adding or setting an entity. It is not possible to define it via `withEntities`.
+
+Therefore, all variations of the `add*` and `set*` functions have an optional (last) parameter, which is an object literal and allows to define the id property via `idKey`.
+
+For example:
+
+```typescript
+interface Todo {
+  key: number;
+  name: string;
+  finished: boolean;
+}
+
+patchState(
+  this.todoStore,
+  addEntities(
+    [
+      { key: 2, name: 'Car Washing', finished: false },
+      { key: 3, name: 'Cat Feeding', finished: false },
+    ],
+    { idKey: 'key' }
+  )
+);
+
+patchState(this.todoStore, setEntity({ key: 4, name: 'Dog Feeding', finished: false }, { idKey: 'key' }));
+```
+
+The `update*` and `remove*` methods, which expect an id value, will automatically pick the right one. That is possible because every entity belongs to a map with its id as the key.
```

Comment:
```suggestion
The `update*` and `remove*` methods, which expect an id value, automatically pick the right one. That is possible because every entity belongs to a map with its id as the key.
```